src/frontend/mame/ui/icorender.cpp

```diff
@@ -212,8 +212,8 @@ int images_in_ico(util::core_file &fp)
 {
 	// read and check the icon file header
 	std::error_condition err;
-	size_t actual;
-	icon_dir_t header;
+	size_t actual = 0;
+	icon_dir_t header{};
```

Comment:
Same comment as the other case with `actual`:
* If `err` is clear at line 218, `fp.read` will set `actual`.
* `actual` will only be checked at line 220 if `err` is clear – you can’t get here with `err` clear if `fp.read` has not been set.

Additionally, default-initialising `header` is unnecessary work when it won’t be used if `fp.read` fails or returns short.

Reply:
Not default initilising it causes MSVC to bail again with a 'warnings as errors' message because it thinks it might be used.